domains/certificates/Shared/DomainCertificate/ShieldedValue.cs

```diff
@@ -0,0 +1,14 @@
+using System.Numerics;
+
+namespace DomainCertificate;
+
+/// <summary>
+/// This class is made to support Pedersen-commitment.
+/// </summary>
+/// <typeparam name="T"></typeparam>
+/// <param name="Shielded">Stamdata</param>
+/// <param name="R">Random number</param>
+public record ShieldedValue<T>(
+    T Shielded, // stamdata
```

Comment:
Renaming this from `Value` to `Shielded` is a breaking change. We have data in the database where the property is named `Value`. We have two options:

1) Revert this back to `Value`
2) Write migration scripts that renames the property from `Value` to `Shielded` on all existing data. 

Regarding option 2), please be aware that we do not have a mechanism for executing migration scripts, so option 2 requires a big amount of work. 
I do not think the rename provides much value anyway, so I vote for option 1).

Reply:
I changed it so shielded value objects are not accessed by value.value. But I see, if it's breaking I'll revert